projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts

```diff
@@ -135,13 +146,17 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
     private _end = 0;
     private _start = 0;
     private _key: string;
+    private _mask: string;
```

Comment:
having `_mask` private field we can remove both `_oldMask` and `defaultMask` now. First initialize `_mask` to be equal to `CCCCCCCCCC` and default may gone. Then in `setPlaceholder` check against `_mask`. To make this work you should also call in `mask` setter first `setPlaceholder` and then set `_mask` to provided value.